tests/platforms/test_host_collector.py

```diff
@@ -184,6 +194,54 @@ def test_prepare_payload_with_snapshot_disabled_python_packages(self, mock_shoul
         self.assertIn('snapshot', payload['metrics']['plugins'][0]['data'])
         snapshot =  payload['metrics']['plugins'][0]['data']['snapshot']
         self.assertTrue(snapshot)
+        self.assertIn('m', snapshot)
+        self.assertEqual('Manual', snapshot['m'])
         self.assertIn('version', snapshot)
         self.assertEqual(len(snapshot['versions']), 1)
         self.assertEqual(snapshot['versions']['instana'], VERSION)
+
+
+    @patch.object(HostCollector, "should_send_snapshot_data")
+    def test_prepare_payload_with_autowrapt(self, mock_should_send_snapshot_data):
+        mock_should_send_snapshot_data.return_value = True
+        os.environ["AUTOWRAPT_BOOTSTRAP"] = "instana"
+        self.create_agent_and_setup_tracer()
+
+        payload = self.agent.collector.prepare_payload()
+        self.assertTrue(payload)
+        self.assertIn('snapshot', payload['metrics']['plugins'][0]['data'])
+        snapshot =  payload['metrics']['plugins'][0]['data']['snapshot']
+        self.assertTrue(snapshot)
+        self.assertIn('m', snapshot)
+        self.assertEqual('Autowrapt', snapshot['m'])
+        self.assertIn('version', snapshot)
+        self.assertGreater(len(snapshot['versions']), 5)
+        self.assertEqual(snapshot['versions']['instana'], VERSION)
+        self.assertIn('wrapt', snapshot['versions'])
+        self.assertIn('fysom', snapshot['versions'])
+        self.assertIn('opentracing', snapshot['versions'])
+        self.assertIn('basictracer', snapshot['versions'])
+
+
+    @patch.object(HostCollector, "should_send_snapshot_data")
+    def test_prepare_payload_with_autotrace(self, mock_should_send_snapshot_data):
+        mock_should_send_snapshot_data.return_value = True
+
+        sys.path.append('/tmp/.instana/python')
+
+        self.create_agent_and_setup_tracer()
+
+        payload = self.agent.collector.prepare_payload()
+        self.assertTrue(payload)
+        self.assertIn('snapshot', payload['metrics']['plugins'][0]['data'])
+        snapshot =  payload['metrics']['plugins'][0]['data']['snapshot']
+        self.assertTrue(snapshot)
+        self.assertIn('m', snapshot)
+        self.assertEqual('AutoTrace', snapshot['m'])
+        self.assertIn('version', snapshot)
+        self.assertGreater(len(snapshot['versions']), 5)
+        self.assertEqual(snapshot['versions']['instana'], VERSION)
+        self.assertIn('wrapt', snapshot['versions'])
+        self.assertIn('fysom', snapshot['versions'])
+        self.assertIn('opentracing', snapshot['versions'])
+        self.assertIn('basictracer', snapshot['versions'])
```

Comment:
Should we loop through the expected libraries like this? (with or without an optional message when it fails)
```suggestion
        expected_packages = ['wrapt', 'fysom', 'opentracing', 'basictracer']
        for package in expected_packages:
            self.assertIn(package, snapshot['versions'], f"{package} not found in snapshot['versions']")

```

Reply:
Valid point, we should have a loop.

Reply:
done